toolchest/pyomoio.py

```diff
@@ -0,0 +1,217 @@
+"""pyomoio: read data from Pyomo models to pandas DataFrames
+
+Pyomo is a AMPL-like model description language for mathematical
+optimization problems. This module provides functions to read data from 
+Pyomo model instances and result objects. Use list_entities to get a list
+of all entities (sets, params, variables, objectives or constraints) inside a 
+pyomo instance, before get its contents by get_entity (or get_entities).
+    
+"""
+import pyomo.core as pyomo
+import pandas as pd
+
+def get_entity(instance, name):
+    """Return a DataFrame for an entity in model instance
+    
+    Parameters
+    ----------
+    instance: a Pyomo ConcreteModel instance
+    name: str, name of a Set, Param, Var, Constraint or Objective in instance
+    
+    Returns
+    -------
+    a single-columned Pandas DataFrame with domain as index
+    """
+
+    # retrieve entity, its type and its onset names
+    entity = instance.__getattribute__(name)
+    labels = _get_onset_names(entity)
+
+    # extract values
+    if isinstance(entity, pyomo.Set):
+        # Pyomo sets don't have values, only elements
+        results = pd.DataFrame([(v, 1) for v in entity.value])
+
+        # for unconstrained sets, the column label is identical to their index
+        # hence, make index equal to entity name and append underscore to name
+        # (=the later column title) to preserve identical index names for both
+        # unconstrained supersets
+        if not labels:
```

Comment:
retracting this comment, TIL it's enshrined in PEP8 that you *should* check emptyness of containers with their implicit bool cast!